src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/delete/AttemptDeleteStep.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.step.delete
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.elasticapi.suspendUntil
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.DeleteActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.Step
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import java.time.Instant
+
+class AttemptDeleteStep(
+    val clusterService: ClusterService,
+    val client: Client,
+    val config: DeleteActionConfig,
+    managedIndexMetaData: ManagedIndexMetaData
+) : Step(name, managedIndexMetaData) {
+
+    private val logger = LogManager.getLogger(javaClass)
+    private var failed: Boolean = false
+    private var info: Map<String, Any>? = null
+
+    // TODO: Incorporate retries from config and consumed retries from metadata
+    // TODO: Needs to return execute status after finishing, i.e. succeeded, noop, failed, failed info to update metadata
+    override suspend fun execute() {
+        val response: AcknowledgedResponse = client.admin().indices()
+                .suspendUntil { delete(DeleteIndexRequest(managedIndexMetaData.index), it) }
+        if (!response.isAcknowledged) {
+            failed = true
+            info = mapOf("message" to "Failed to delete index")
+        }
+    }
+
+    override fun getUpdatedManagedIndexMetaData(currentMetaData: ManagedIndexMetaData): ManagedIndexMetaData {
+        return currentMetaData.copy(
+            step = name,
+            stepStartTime = getStepStartTime().toEpochMilli().toString()
+            // transitionTo = null
+            // stepComplete = !failed
+            // failed = failed
+            // info = info
+        )
+    }
+
+    companion object {
```

Comment:
I don't now if we have best practice around this but I feel like it is easier to read `companion object` near the top of the class because I think these variables are considered `static` in kotlin. So having them at the top makes it easier to read.

Reply:
Current recommendation is to keep it at the bottom
https://kotlinlang.org/docs/reference/coding-conventions.html#class-layout